components/sup/src/manager/service/health.rs

```diff
@@ -40,12 +46,177 @@ impl From<i8> for HealthCheck {
 
 impl fmt::Display for HealthCheck {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let msg = match *self {
-            HealthCheck::Ok => "OK",
-            HealthCheck::Warning => "WARNING",
-            HealthCheck::Critical => "CRITICAL",
-            HealthCheck::Unknown => "UNKNOWN",
-        };
-        write!(f, "{}", msg)
+        match *self {
+            HealthCheck::Ok => "OK".fmt(f),
+            HealthCheck::Warning => "WARNING".fmt(f),
+            HealthCheck::Critical => "CRITICAL".fmt(f),
+            HealthCheck::Unknown => "UNKNOWN".fmt(f),
+        }
+    }
+}
+
+/// All state needed for checking the health of a service over time.
+#[derive(Clone)]
+pub struct State {
+    // All hooks currently need these
+    hook:                   Option<Arc<HealthCheckHook>>,
+    service_group:          ServiceGroup,
+    package:                Pkg,
+    svc_encrypted_password: Option<String>,
+
+    /// A reference to the process supervisor for the service. This is
+    /// used to create a "proxy health check" for services that do not
+    /// provide their own health check hook.
+    supervisor: Arc<Mutex<Supervisor>>,
+
+    /// The configured interval at which to run health checks for this
+    /// service. The interval actually used may differ based on the
+    /// previous health check result.
+    configured_interval: HealthCheckInterval,
```

Comment:
```suggestion
    nominal_interval: HealthCheckInterval,
```
May be a slightly more illustrative name. To me, "configured" implies that it's been explicitly set, whereas this will usually just be the default value of 30 seconds. Nominal focuses on the fact that it's the interval applied in *normal* operation, and that erroneous circumstances may use a different one.

Reply:
I like it!